### DISCUSSION TITLE
Scan until EAN found?

### DISCUSSION THREAD

**Discussion by gtzippy** [journeyapps/zxing-android-embedded#732] [Q&A]
I'm using this for a little home project with my kid's books. It is great, and easy for a young child to use, but sometimes the scanner stops and returns the barcode before it finds the EAN suffix. Is there a way to say "do not stop scanning a barcode until you find an ean5 or ean2 suffix"?

**Comment by rkistner** [comment]
It's currently not possible via the default method of using ScanContract & CaptureActivity. But if you use a custom Activity, you can replace the DecoderFactory using this:

https://github.com/journeyapps/zxing-android-embedded/blob/d09b7c76c3124fbfbd096a65d60b1997f37ff90f/zxing-android-embedded/src/com/journeyapps/barcodescanner/BarcodeView.java#L103

Then you can set this hint on the decoder:

https://github.com/zxing/zxing/blob/dfb06fa33b17a9e68321be151c22846c7b78048f/core/src/main/java/com/google/zxing/DecodeHintType.java#L92-L99

Another option is to use the continuous scanning example as a starting point, and only stop scanning if you find a barcode with the suffix.
